src/commonMain/kotlin/com/configcat/ConfigCatClient.kt

```diff
@@ -202,79 +338,117 @@ internal class Client private constructor(
     }
 
     override suspend fun getAllKeys(): Collection<String> {
-        return getSettings().keys
+        return getSettings().settings.keys
     }
 
     override suspend fun getAllValues(user: ConfigCatUser?): Map<String, Any> {
-        return getSettings().map {
-            val evaluated = evaluator.evaluate(it.value, it.key, user)
-            it.key to evaluated.first
+        val result = getSettings()
+        return result.settings.map {
+            val evaluated = evaluate(it.value, it.key, user ?: defaultUser, result.fetchTime)
+            it.key to evaluated.value
         }.toMap()
     }
 
     override suspend fun getAllVariationIds(user: ConfigCatUser?): Collection<String> {
-        return getSettings().map {
-            val evaluated = evaluator.evaluate(it.value, it.key, user)
-            evaluated.second
+        val result = getSettings()
+        return result.settings.map {
+            val evaluated = evaluate(it.value, it.key, user ?: defaultUser, result.fetchTime)
+            evaluated.variationId
         }.filterNotNull()
     }
 
-    override suspend fun refresh() {
-        service?.refresh()
+    override suspend fun forceRefresh(): RefreshResult = service?.refresh() ?: RefreshResult(false,
+        "The ConfigCat SDK is in local-only mode. Calling .forceRefresh() has no effect.")
```

Comment:
Could you please add warning log here:

offline mode warning (eg: the client is OFFLINE and ForceRefresh was invoked)
Level: WARNING
Message: Client is in offline mode, it can't initiate HTTP calls.

Reply:
Done